platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -302,6 +302,11 @@ private boolean shouldUseInitProjectSpec(UserInputHandler inputHandler) {
     private void doInitSpecProjectGeneration(UserInputHandler inputHandler) {
         BuildInitConfig config = inputHandler.askUser(this::selectAndConfigureSpec).get();
         BuildInitGenerator generator = createGenerator(config);
+        boolean userInterrupted = inputHandler.interrupted();
+        if (userInterrupted) {
+            throw new BuildCancelledException();
+        }
```

Comment:
💭 This only cancels generation prior to the generator actually being run though.  

If they cancel it during generation (`generator.generate(config, projectDir)`), nothing.  During generation is where I'd expect to spend most of the time in this process, once the init specs get dynamic.

Also, this doesn't look like it handles cancellation during any (to be written) back and forth about build init parameter question-asking.

So I guess what I'm saying is...why not wrap the entire body of this method in the `try-catch`?

Reply:
Yes, there's an issue with how this would work when we start asking questions in the generator, but we don't have that yet.

There isn't a way to try/catch around this because the interrupted signal can be set any time. The way the normal path handles this is that it asks all the questions first, then it checks if the user cancelled and then it generates the files. When the user cancels, we actually treat that as answering all the questions with the default answer.

I think that means we'd need to change the generator a bit so that we are in control when all of the questions have been asked so we can check for cancellation before allowing the generator to run.

Reply:
OK.  Something to think about when we implement the parameterization of later build init specs.